### PR TITLE
Remove ansi escape chars in assertExpectedInline and add options to skip comments and to skip empty lines

### PR DESCRIPTION
**Pull request by laithsakka**
I had a night mare rewriting tests in test_misc.py specifically :
1. graphs can have comments that refers to my files "/lsakka/.." we really dont care about comments add option to ignore comments. 
2. empty lines added when EXPECTTEST_ACCEPT=1  are changed with linter causing tests to fail or linter fail!
add flag to ignore empty lines. 
3. EXPECTTEST_ACCEPT fails when the text have some not readable characters. those should not effect comparing strings, also those causes weird diffs comments when tests fails. I removed ansi_escape chars https://github.com/pytorch/pytorch/pull/133045

this is used in  
 
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #134409
* __->__ #134248
* #134364
* #133639

### CODE REVIEW

**Review by aorenste — COMMENTED** [review]
Is this actually better than finding where the escape codes are coming from and adding an option to turn them off?
(Where are they coming from? Something internal or external?)

**Review by oulgen — CHANGES_REQUESTED** [review]
This comes from `colored` argument to `_gen_python_code`

https://github.com/pytorch/pytorch/blob/0eb9c870fdaa351d5df8aa7aec6eae07baaa7ee5/torch/fx/graph.py#L379

Can you just set it false for unit tests?

**Comment by laithsakka** [comment]
> This comes from `colored` argument to `_gen_python_code`
> 
> https://github.com/pytorch/pytorch/blob/0eb9c870fdaa351d5df8aa7aec6eae07baaa7ee5/torch/fx/graph.py#L379
> 
> Can you just set it false for unit tests?

we can but why not making the tool incentive to those?

**Comment by laithsakka** [comment]
> Is this actually better than finding where the escape codes are coming from and adding an option to turn them off? (Where are they coming from? Something internal or external?)

yeh i think its better, since when comparing strings anyway those are not taken into account.

File: torch/testing/_internal/common_utils.py [hunk 1]
```diff
@@ -2860,7 +2860,44 @@ def assertLeaksNoCudaTensors(self, name=None):
     def enforceNonDefaultStream(self):
         return CudaNonDefaultStream()
 
-    def assertExpectedInline(self, actual, expect, skip=0):
+    def _remove_ansi_escape(self, input):
+        # 7-bit C1 ANSI sequences
+        ansi_escape = re.compile(r'''
```
Comment (aorenste) [inline torch/testing/_internal/common_utils.py:2865]: I dislike the fact that this regex uses sequences that cross chartype boundaries (like "@-Z") but in general is reasonable.

**Review by aorenste — APPROVED** [review]
Approving - I think this isn't unreasonable.  In general our tools shouldn't be unconditionally emitting ansi codes in their output, but maybe that's a separate battle.

File: torch/testing/_internal/common_utils.py (hunk 1 above)
Comment (laithsakka) [reply torch/testing/_internal/common_utils.py:2865]: alternatively I can add a flag ignore_ansi_escape, like the other two flags.

**Pull request closed without merging**